internal/format/format.go

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0'
+ */
+
+package format
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/edgexfoundry/edgex-cli/internal/config"
+	"github.com/edgexfoundry/edgex-cli/internal/service"
+)
+
+type FormatType int
+
+const (
+	Debug FormatType = iota
+	JSON
+	Value
+)
+
+func PrintServiceTable(endpoint string, ft FormatType, jsonValueName string) {
+	PrintServices(config.Configuration.CoreServices, endpoint, ft, jsonValueName)
+	PrintServices(config.Configuration.DeviceServices, endpoint, ft, jsonValueName)
+	PrintServices(config.Configuration.ApplicationServices, endpoint, ft, jsonValueName)
+
+}
+
+func PrintServices(service config.Services, endpoint string, ft FormatType, jsonValueName string) {
+	for serviceName, service := range service {
+
+		var value, jsonValue, url string
+		var err error
+		value, jsonValue, url, err = GetServiceValue(service, endpoint, jsonValueName)
+		if err != nil {
+			if ft == Debug {
+				fmt.Printf("%s: %s: %s\n", serviceName, url, err.Error())
+			}
+		} else {
+			switch ft {
+
+			case Debug:
+				fmt.Printf("%s: %s: %s\n", serviceName, url, jsonValue)
+
+			case JSON:
+				fmt.Printf("%s: %s\n", serviceName, jsonValue)
+			case Value:
+				if strings.Contains(jsonValueName, "\n") {
+					fmt.Printf("%s:\n%s\n", serviceName, value)
+				} else {
+					fmt.Printf("%s: %s\n", serviceName, value)
+				}
+			default:
+				fmt.Println("Unknown type")
+			}
+		}
+	}
+}
+
+func GetServiceValue(s service.Service, endpoint string, jsonValueName string) (value string, jsonStr string, url string, err error) {
```

Comment:
All public funcs/structs should have Godoc style comments describing the func/struct

Reply:
Also, maybe a more descriptive function name. From the code , I can't determine the purpose of the function.

Reply:
Good point! I'll update all of them.